core/DataAccess/LogAggregator.php

```diff
@@ -1236,9 +1236,9 @@ public function queryConversionsByPageView(string $linkField, int $idGoal)
             " . sprintf("ROUND(SUM(1 / log_conversion.pageviews_before * log_conversion.revenue_tax),2) AS `%d`,", Metrics::INDEX_GOAL_ECOMMERCE_REVENUE_TAX) . "
             " . sprintf("ROUND(SUM(1 / log_conversion.pageviews_before * log_conversion.revenue_shipping),2) AS `%d`,", Metrics::INDEX_GOAL_ECOMMERCE_REVENUE_SHIPPING) . "
             " . sprintf("ROUND(SUM(1 / log_conversion.pageviews_before * log_conversion.revenue_discount),2) AS `%d`,", Metrics::INDEX_GOAL_ECOMMERCE_REVENUE_DISCOUNT) . "
-            " . sprintf("SUM(1 / log_conversion.pageviews_before * log_conversion.items) AS `%d`,", Metrics::INDEX_GOAL_ECOMMERCE_ITEMS) . "
+            " . sprintf("SUM(ROUND(1 / log_conversion.pageviews_before * log_conversion.items, 4)) AS `%d`,", Metrics::INDEX_GOAL_ECOMMERCE_ITEMS) . "
```

Comment:
Should we round the result only instead of each value? Similar to how other columns are done above and below? Also we use just 2 decimal places there.
```suggestion
            " . sprintf("ROUND(SUM(1 / log_conversion.pageviews_before * log_conversion.items), 4) AS `%d`,", Metrics::INDEX_GOAL_ECOMMERCE_ITEMS) . "
```

Reply:
The number is currently used with 4 fraction digits in the reports, so limiting it to 2 is sort of a breaking change.
Also there is a slightly difference between `SUM(ROUND())` and `ROUND(SUM())`. Summing rounded values ensures that the results are the same everywhere, as a precision of 4 is supported on all engines. If you first sum the values, the result can differ depending on the precision, and even if unlikely and round afterwards could have a different result.

Reply:
I get the difference between summing rounded values or rounding the sum, but isn't the same issue then relevant to the other columns in the query that are summed first and then rounded?

Reply:
In theory yes. But it's more unlikely with less fraction digits. Also the other numbers don't make any problems with tests on TiDb, so I wanted to keep the changes minimal.

Reply:
Fair enough. I wanted to avoid the head scratching later on when someone looks at the query and will have no clue why one is round(sum( and the other is sum(round(.

Reply:
But if we switch it around for the other columns and the results in tests don't change, we can make it consistent for all columns on the order of sum/round, maybe?
Otherwise we should at least create a ticket to follow up.

Reply:
Tried changing the other columns, and some tests are starting to fail. Guess the only thing that would work would be something like `ROUND(SUM(ROUND(1 / column, 4)),2)`, so that the initial calculation uses the same amount of fraction digits, but the final number is rounded to less fraction digits then.

Reply:
@caddoo @mneudert do you want to weight in here or leave it as is?

Reply:
As discussed lets just leave it as is for now, a ticket will be created by @sgiehl as a follow up